src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -64,13 +70,15 @@
      * @param root {@code Node} root of the tree
      * @param parent {@code Node} parent of the current node
      * @param detailAst reference to {@code DetailAST}
+     * @param position position of the current node among its siblings
      */
-    public ElementNode(AbstractNode root, AbstractNode parent, DetailAST detailAst) {
+    public ElementNode(AbstractNode root, AbstractNode parent, DetailAST detailAst, int position) {
         super(root.getTreeInfo());
         this.parent = parent;
         this.root = root;
         this.detailAst = detailAst;
         text = TokenUtil.getTokenName(detailAst.getType());
+        positionAmongSiblings = position;
```

Comment:
Can we get index from parent node ? `parent.getChildren().size() - 1`

It is better to name field as `index...` to imply that it is strict sequence